pkgs/dart_tooling_mcp_server/lib/src/server.dart

```diff
@@ -5,34 +5,47 @@
 import 'dart:async';
 
 import 'package:dart_mcp/server.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
 import 'package:stream_channel/stream_channel.dart';
 
 import 'mixins/analyzer.dart';
 import 'mixins/dart_cli.dart';
 import 'mixins/dtd.dart';
+import 'utils/process_manager.dart';
 
 /// An MCP server for Dart and Flutter tooling.
 final class DartToolingMCPServer extends MCPServer
     with
+        ProcessManagerSupport,
         LoggingSupport,
         ToolsSupport,
         DartAnalyzerSupport,
         DartCliSupport,
         DartToolingDaemonSupport {
-  DartToolingMCPServer({required super.channel})
-    : super.fromStreamChannel(
-        implementation: ServerImplementation(
-          name: 'dart and flutter tooling',
-          version: '0.1.0-wip',
-        ),
-        instructions:
-            'This server helps to connect Dart and Flutter developers to '
-            'their development tools and running applications.',
-      );
+  DartToolingMCPServer({
+    required super.channel,
+    @visibleForTesting
+    LocalProcessManager processManager = const LocalProcessManager(),
+  }) : super.fromStreamChannel(
+         implementation: ServerImplementation(
+           name: 'dart and flutter tooling',
+           version: '0.1.0-wip',
+         ),
+         instructions:
+             'This server helps to connect Dart and Flutter developers to '
+             'their development tools and running applications.',
+       ) {
+    _processManager = processManager;
+  }
 
   static Future<DartToolingMCPServer> connect(
     StreamChannel<String> mcpChannel,
   ) async {
     return DartToolingMCPServer(channel: mcpChannel);
   }
+
+  @override
+  LocalProcessManager get processManager => _processManager;
+  late final LocalProcessManager _processManager;
```

Comment:
```suggestion
  final LocalProcessManager processManager;
```

Plus the associated change above